.travis.yml

```diff
@@ -61,6 +61,17 @@ matrix:
           packages:
             - gcc-multilib
             - libgmp-dev:i386
+    # clang with memory sanitizer
+    - compiler: clang
+      # --disable-openssl-tests because openssl uses uninitialized memory. ASM
+      # and BIGNUM are disabled because clang memory sanitizer does not work
+      # with inline assembly (https://clang.llvm.org/docs/MemorySanitizer.html).
+      # The memory sanitizer is instructed to exit with a different exit code
+      # using MSAN_OPTIONS. This is because the default exit code is 77 - the
+      # same exit code that autotools make check interprets as a test that is
+      # supposed to be skipped.
+      env: EXTRAFLAGS="--disable-openssl-tests CFLAGS=-fsanitize=memory" ASM=no BIGNUM=no EXPERIMENTAL=yes ENDOMORPHISM=yes RECOVERY=yes ECDH=yes MSAN_OPTIONS=exitcode=42
```

Comment:
Maybe it's a good idea to test with both `ENDOMORPHISM=yes` and `ENDOMORPHISM=no` because both code paths are interesting for memory: setting up the contexts involves a lot of pointer arithmetic etc.

Reply:
Yeah I can add that.